### PR TITLE
Impala: add support for VALUES aliases

### PR DESCRIPTION
**Pull request by strang3nt**
### Brief summary of the change made

Updates both Hive and Impala dialects. Hive:

- ~Added DEFAULT as unreserved keyword (see docs at <https://hive.apache.org/docs/latest/language/languagemanual-ddl/#keywords-non-reserved-keywords-and-reserved-keywords>)~
- ~improved VALUES clause:~
  - ~allow DEFAULT clauses in INSERT (supported from version 3.0.0, see <https://issues.apache.org/jira/browse/HIVE-18726> and <https://issues.apache.org/jira/browse/HIVE-19059>).~
  - ~allow aliases in VALUES clauses, only in first row, couldn't find documentation tried in hive container instead.~

~Commands on a Hive container:~
 
- ~[run_hive.sh](https://github.com/user-attachments/files/30202726/run_hive.sh)~
- ~[hive_output.txt](https://github.com/user-attachments/files/30202727/hive_output.txt)~
- ~[values_as.sql](https://github.com/user-attachments/files/30202728/values_as.sql)~

Impala:

- Support for aliases in VALUES, ~except when in INSERT clauses~ (see here for aliasing <https://impala.apache.org/docs/build/html/topics/impala_values.html> and here <https://impala.apache.org/docs/build/html/topics/impala_insert.html> for supported insert syntax).

### Are there any other side effects of this change that we should be aware of?

No

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - [x] `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

---
## Summary by cubic
Adds Impala support for aliases in VALUES tuples (with DEFAULT) across SELECT/UNION and INSERT, and adds focused parser tests. Also simplifies the Impala grammar; Hive behavior remains as previously updated.

- **New Features**
  - Hive: `DEFAULT` is unreserved; allowed only in INSERT. Aliases allowed only in the first `VALUES` row.
  - Impala: `VALUES` supports aliases and `DEFAULT` in both SELECT/UNION and INSERT. Added fixtures validating both contexts.

- **Refactors**
  - Impala: Extend `ansi.ValuesClauseSegment` and consolidate match grammar with greedy parsing to reduce complexity.

<sup>Written for commit 1e57442b6b013945f4516669c1b2d131fb257acc. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/sqlfluff/sqlfluff/pull/8185?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

**Review by peterbud — CHANGES_REQUESTED** [review]
Hi @strang3nt - thanks for this PR and welcome!

I’m not comfortable with this as-is. The change is broader than the PR description suggests, because it replaces the normal `VALUES` handling with dialect-local grammar in both dialect_hive.py and dialect_impala.py, rather than making a narrow fix to the specific unsupported cases.

Two concrete problems showed up when I checked the parser behavior. In Hive, the `FROM ... INSERT OVERWRITE DIRECTORY ... SELECT ... INSERT ...` chaining regresses because the `SELECT` arm no longer stops cleanly at the next `INSERT`. Also in Hive, the split between select-values and insert-values doesn’t really enforce “DEFAULT only in INSERT”, because `DEFAULT` still parses through the generic expression path in `UNION VALUES`.

More generally, wiring `InsertValuesClauseSegment` into Hive `FROM ... INSERT` paths looks semantically wrong for multi-insert syntax, and the new tests only cover the happy-path examples, not the boundary cases that would catch these regressions. I’d strongly prefer a smaller change that preserves the existing ANSI/Selectable behavior and adds explicit negative tests for the unsupported forms, or even separate the two dialect changes into separate PRs.

**Comment by strang3nt** [comment]
Hi @peterbud, thank you for the detailed feedback! On it :eyes:. I'm going to try and reduce the scope of the change and add negative test cases.

> More generally, wiring `InsertValuesClauseSegment` into Hive `FROM ... INSERT` paths looks semantically wrong for multi-insert syntax, and the new tests only cover the happy-path examples, not the boundary cases that would catch these regressions. I’d strongly prefer a smaller change that preserves the existing ANSI/Selectable behavior and adds explicit negative tests for the unsupported forms, or even separate the two dialect changes into separate PRs.

Regarding splitting the PR: I'd like to try narrowing the implementation first. If the change still looks too messy after that, I'm fully open to splitting it into separate PRs

**Comment by strang3nt** [comment]
Hi @peterbud,

After a bit more investigation, I realized the changes required for the Impala dialect are simpler than I originally thought. If you are still open to it, I would like to split this into two PRs:

- This PR: Add DEFAULT as an unreserved keyword for Hive (which I now realize can be used anywhere, though it may yield a semantic error) and override `ValuesClauseSegment` for Impala to add support for aliasing (as per [e4d481c](https://github.com/sqlfluff/sqlfluff/pull/8185/commits/e4d481c667d15cddae3336f5457a303f3a743427)).
- Next PR: Iron out the Hive changes, which seem a bit more nuanced and have little documentation to support them.

If you agree, I'll update the PR description and revert (or rebase) the extra Hive changes.

File: src/sqlfluff/dialects/dialect_impala.py [hunk 1]
```diff
@@ -35,6 +38,38 @@
...
+                Bracketed(
+                    Delimited(
+                        "DEFAULT",
+                        Sequence(
+                            OneOf(
+                                "DEFAULT",
+                                Ref("LiteralGrammar"),
+                                Ref("ExpressionSegment"),
+                            ),
+                            Ref("AliasExpressionSegment"),
```
[the remark this answers is not in the document]
Comment (strang3nt) [reply src/sqlfluff/dialects/dialect_impala.py:61]: Note that syntax described [here](https://impala.apache.org/docs/build/html/topics/impala_values.html) allows aliases on every row

**Comment by peterbud** [comment]
Hi @strang3nt - sure, go ahead, let's split this PR into two pieces.

**Comment by peterbud** [comment]
As I see there is no Hive change in the PR anymore, only the Impala - is that intentional?

**Comment by strang3nt** [comment]
Thanks for updating the PR description. I forgot about adding the DEFAULT keyword, but grouping all Hive-related changes into a separate PR makes more sense.

**Review by peterbud — APPROVED** [review]
Thanks!
